rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java

```diff
@@ -134,6 +157,299 @@ public String getCyclicLinkedProjectPath(Project project, User user) throws TExc
+        }
+
+        return mergedUsage;
+    }
+
+    public Map<String, ObligationStatusInfo> getLicenseObligationData(Map<String, Set<Release>> licensesFromAttachmentUsage, User user) {
+        ThriftClients thriftClients = new ThriftClients();
+        LicenseService.Iface licenseClient = thriftClients.makeLicenseClient();
+        Map<String, ObligationStatusInfo> obligationStatusMap = new HashMap<String, ObligationStatusInfo>();
+        licensesFromAttachmentUsage.entrySet().stream().forEach(entry -> wrapTException(() -> {
+            License lic = null;
+            Set<Release> releaseData = entry.getValue();
+            Map<String, String> releaseIdToAcceptedCli = new HashMap<String, String>();
+            for (Release rel : releaseData) {
+                String releaseId = rel.getId();
+                for (Attachment attachment : rel.getAttachments()) {
+                    if (CheckStatus.ACCEPTED.equals(attachment.getCheckStatus())) {
+                        String attachmentContentId = attachment.getAttachmentContentId();
+                        releaseIdToAcceptedCli.put(releaseId, attachmentContentId);
+                    }
+                }
+            }
+            try {
+                lic = licenseClient.getByID(entry.getKey(), user.getDepartment());
+            } catch (TException exp) {
+                log.warn("Error fetching license from backend! License Id-" + entry.getKey(), exp.getMessage());
+                return;
+            }
+            if (lic == null || CommonUtils.isNullOrEmptyCollection(lic.getObligations()))
+                return;
+
+            lic.getObligations().stream().filter(Objects::nonNull).forEach(obl -> {
+                String keyofObl = CommonUtils.isNotNullEmptyOrWhitespace(obl.getTitle()) ? obl.getTitle()
+                        : obl.getText();
+                ObligationStatusInfo osi = null;
+                if (obligationStatusMap.containsKey(keyofObl)) {
+                    osi = obligationStatusMap.get(keyofObl);
+                } else {
+                    osi = new ObligationStatusInfo();
+                    obligationStatusMap.put(keyofObl, osi);
+                }
+                osi.setText(obl.getText());
+                osi.setId(obl.getId());
+                osi.setObligationType(obl.getObligationType());
+                osi.setObligationLevel(obl.getObligationLevel());
+                osi.setReleaseIdToAcceptedCLI(releaseIdToAcceptedCli);
+                Set<String> licenseIds = osi.getLicenseIds();
+                if (licenseIds == null) {
+                    licenseIds = new HashSet<>();
+                    osi.setLicenseIds(licenseIds);
+                }
+                licenseIds.add(entry.getKey());
+            });
+
+        }));
+        return obligationStatusMap;
+    }
+
+    public RequestStatus updateLinkedObligations(Project project, User user, Map<String, ObligationStatusInfo> licenseObligation) throws TException {
```

Comment:
**TException** is never thrown

Reply:
Used on line 367.

Reply:
@rudra-superrr, but you wrapped your code snippet in a try-catch block, so I don't think this function can throw Exception

Reply:
Understood.